src/Operation/FindAndModify.php

```diff
@@ -243,6 +257,16 @@ public function execute(Server $server)
 
         $cursor = $server->executeWriteCommand($this->databaseName, new Command($this->createCommandDocument()), $this->createOptions());
 
+        if (isset($this->options['codec'])) {
+            $cursor->setTypeMap(['root' => 'bson']);
+            $result = current($cursor->toArray());
+            assert($result instanceof Document);
+
+            $decoded = $this->options['codec']->decode($result->get('value'));
```

Comment:
Note that below we gracefully handle `value` being unset, which is possible depending on the `new` option and whether something was matched. May be worth adding a functional regression test for this since if it wasn't caught already by a test failure.

Reply:
@alcaeus: I think this is still outstanding. The method is still doing:

```php
return $this->options['codec']->decode($result->get('value'));
```

Reply:
My apologies, this got lost as I changed everything to `decode` instead of `decodeIfSupported` and my brain got confused. I changed the logic to directly return `null` values, but to use `decode` for all other values. This ensures that the result will always be of the type indicated by the codec, or `null`. Using `decodeIfSupported` could mean that we're returning a `Document` instance which may be unexpected.

Reply:
The most recent version in this PR is doing the following for the `codec` code path:

```php
$value = $result->get('value');

return $value === null ? $value : $this->options['codec']->decode($value);
```

This doesn't address my original comment (https://github.com/mongodb/mongo-php-library/pull/1140#discussion_r1272534094) about `value` being unset; however, I actually think that's no longer a concern with MongoDB 3.6+. The original code path where we handle a missing `value` property may have been related to some Very Old™ code ([PHPLIB-95](https://jira.mongodb.org/browse/PHPLIB-95) and [SPEC-201](https://jira.mongodb.org/browse/SPEC-201).

The [MongoDB 3.6 documentation for `findAndModify`](https://www.mongodb.com/docs/v3.6/reference/command/findAndModify/#findandmodify-value-field) suggests that `value` is always present but may be `null`.

Feel free to resolve this after reading.